micro-rdk/src/esp32/tls.rs

```diff
@@ -42,6 +42,8 @@ pub struct Esp32TLS {
     #[allow(dead_code)]
     alpn_ptr: Vec<*const c_char>,
     tls_cfg: Either<Box<esp_tls_cfg_server>, Box<esp_tls_cfg>>,
+    #[allow(dead_code)]
+    cfg: Option<Box<Esp32TLSServerConfig>>,
```

Comment:
We need to keep the `Esp32TLSServerConfig` around, because it holds the memory with the cert/key material. Again, there is probably a better way to do this, but getting it onto the heap with `Box` was sufficient to prevent it getting dropped and/or moved as it was before.